### PR TITLE
Remote id UI refactor

### PR DESCRIPTION
**Pull request by bytesByHarsh**
Description
-----------
- Update settings page for remoteID
- Update toolbar indicator for remoteID 

Test Steps
-----------

1) App Settings --> MockLink --> Start Mocklink
2) App Settings --> Remote ID

Related Issue
-----------
https://github.com/mavlink/qgroundcontrol/issues/12018

### CODE REVIEW

**Comment by bytesByHarsh** [comment]
I have updated just the settings page currently, if the changes are ok then I'll remove them from the toolbar UI as well.

**Review by DonLakeFlyer — CHANGES_REQUESTED** [review]
I need to take another look at how validation errors are displayed since this border color changes stuff isn't really a ui standard in QGC. I'll create an issue for that and figure out whaat to do.

Other than the one change this all looks like. Fix that, and I'll merge.

**Comment by bytesByHarsh** [comment]
> I need to take another look at how validation errors are displayed since this border color changes stuff isn't really a ui standard in QGC. I'll create an issue for that and figure out what to do.


@DonLakeFlyer So in the current UI should I completely remove the border color checks and changes?

File: src/UI/preferences/RemoteIDSettings.qml [hunk 1]
```diff
@@ -254,692 +238,395 @@ Rectangle {
...
+                        text:                   operatorIDFact.valueString
+                        visible:                operatorIDFact.visible
+                        maximumLength:          20                  // Maximum defined by Mavlink definition of OPEN_DRONE_ID_OPERATOR_ID message
+
+                        onTextChanged: {
+                            operatorIDFact.value = text
+                            if (_activeVehicle) {
+                                _remoteIDManager.checkOperatorID(text)
+                            } else {
+                                _remoteIDManager.checkOperatorID(text)
```
[the remark this answers is not in the document]
Comment (bytesByHarsh) [reply src/UI/preferences/RemoteIDSettings.qml]: I have corrected the variable and verified it with the toolbar code.

**Pull request merged**